pkg/operator/observe_config.go

```diff
@@ -124,31 +143,30 @@ func observeEtcdEndpoints(kubeClient kubernetes.Interface, clientConfig *rest.Co
 	if len(etcdURLs) > 0 {
 		unstructured.SetNestedStringSlice(observedConfig, etcdURLs, "storageConfig", "urls")
 	} else {
-		glog.Warningf("no etcd endpoints found")
+		return observedConfig, []error{fmt.Errorf("no etcd endpoints found")}
 	}
 	return observedConfig, nil
 }
 
 // observeClusterConfig observes CIDRs from cluster-config-v1 in order to populate list of restrictedCIDRs
-func observeClusterConfig(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, error) {
+func observeClusterConfig(kubeClient kubernetes.Interface, clientConfig *rest.Config, observedConfig map[string]interface{}) (map[string]interface{}, []error) {
+
 	clusterConfig, err := kubeClient.CoreV1().ConfigMaps("kube-system").Get("cluster-config-v1", metav1.GetOptions{})
 	if errors.IsNotFound(err) {
-		glog.Warningf("cluster-config-v1 not found in the kube-system namespace")
-		return observedConfig, nil
+		return observedConfig, []error{fmt.Errorf("cluster-config-v1 not found in the kube-system namespace")}
```

Comment:
maybe worth including the resource type in the error message? (e.e `configmap cluster-config-v1...`)

Reply:
I've added this info to all messages.